eos-tech-support/eos-write-live-image

```diff
@@ -0,0 +1,284 @@
+
+Developer options (you probably don't want to use these):
+   -n,--ntfs               Format the image partition as NTFS, not exFAT
+   -m,--mbr                Format DEVICE as MBR, not GPT (implies --ntfs)
+EOF
+}
+
+function check_exists() {
+    if [ ! -f "$1" ]; then
+        echo "$2 $1 does not exist or is not a file" >&2
+        exit 1
+    fi
+}
+
+NTFS=false
+MBR=false
+WRITABLE=false
+FORCE=false
+while true; do
+    case "$1" in
+        -o|--os-image)
+            shift
+            OS_IMAGE="$1"
+            shift
+            ;;
+        -x|--windows-tool)
+            shift
+            WINDOWS_TOOL="$1"
+            shift
+            ;;
+        -l|--latest)
+            shift
+            FETCH_LATEST=true
+            ;;
+        -n|--ntfs)
+            shift
+            NTFS=true
+            ;;
+        -m|--mbr)
+            shift
+            MBR=true
+            NTFS=true
+            ;;
+        -w|--writable)
+            shift
+            WRITABLE=true
+            ;;
+        -f|--force)
+            shift
+            FORCE=true
+            ;;
+        -h|--help)
+            usage
+            exit 0
+            ;;
+        --)
+            shift
+            break
+            ;;
+    esac
+done
+
+if [ $# -lt 1 ] ; then
+    echo "Missing DEVICE" >&2
+    usage >&2
+    exit 1
+fi
+
+DEVICE="$1"
+
+USERID=$(id -u)
+if [ "$USERID" != "0" ]; then
+    echo "Program requires superuser privileges" >&2
+    exit 1
+fi
+
+# Check for required tools
+if $NTFS; then
+    MKFS_IMAGES="mkfs.ntfs"
+    MKFS_ARGS='--quick -L'
+    MKFS_PACKAGE=ntfs-3g
+else
+    MKFS_IMAGES="mkfs.exfat"
+    MKFS_ARGS=-n
+    MKFS_PACKAGE=exfat-utils
+fi
+
+command -v $MKFS_IMAGES >/dev/null || {
+    echo "$MKFS_IMAGES is not installed... aborting!" >&2
+    echo "Try 'sudo apt-get install $MKFS_PACKAGE'" >&2
+    exit 1
+}
+
+command -v dd >/dev/null || {
+    echo "dd is not installed... aborting!" >&2
+    echo "Try 'sudo apt-get install coreutils'" >&2
+    exit 1
+}
+
+command -v unzip >/dev/null || {
+    echo "unzip is not installed... aborting!" >&2
+    echo "Try 'sudo apt-get install unzip'" >&2
+    exit 1
+}
+
+if [ ! -b $DEVICE ]; then
+    echo "$DEVICE does not exist or is not a block device... aborting!" >&2
+    exit 1
+fi
+
+if grep -qs $DEVICE /proc/mounts; then
+    # Protect against overwriting the device currently in use
+    echo "$DEVICE is currently in use -- please unmount and try again" >&2
+    exit 1
+fi
+
+if ! $FORCE; then
+    read -p "Are you sure you want to overwrite all data on $DEVICE? [y/N] "
+    response=${REPLY,,} # to lower
+    if [[ ! $response =~ ^(yes|y)$ ]]; then
+        exit 1
+    fi
+fi
+
+if [ ! -z "$FETCH_LATEST" ]; then
+    if [ -z "$OS_IMAGE" ]; then
+        OS_IMAGE=$($EOS_DOWNLOAD_IMAGE --product eos)
+    fi
+    if [ -z "$WINDOWS_TOOL" ]; then
+        WINDOWS_TOOL=$($EOS_DOWNLOAD_IMAGE --windows-tool)
+    fi
+fi
+
+if [ ! -f "$WINDOWS_TOOL" -o ! -f "$OS_IMAGE" ]; then
```

Comment:
Use `if [ condition_a ] || [ condition_b ]; then` instead of `-o` since it's more readable that way.